src/_pytest/scope.py

```diff
@@ -0,0 +1,86 @@
+"""
+Scope definition and related utilities.
+
+Those are defined here, instead of in the 'fixtures' module because
+their use is spread across many other pytest modules, and centralizing it in 'fixtures'
+would cause circular references.
+
+Also this makes the module light to import, as it should.
+"""
+from enum import Enum
+from typing import Optional
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing_extensions import Literal
+
+    _ScopeName = Literal["session", "package", "module", "class", "function"]
+
+
+class Scope(Enum):
+    """
+    Represents one of the possible fixture scopes in pytest.
+
+    Scopes are ordered from lower to higher, that is:
+
+              ->>> higher ->>>
+
+    Function < Class < Module < Package < Session
+
+              <<<- lower  <<<-
+    """
+
+    # Scopes need to be listed from lower to higher.
+    Function: "_ScopeName" = "function"
+    Class: "_ScopeName" = "class"
+    Module: "_ScopeName" = "module"
+    Package: "_ScopeName" = "package"
+    Session: "_ScopeName" = "session"
+
+    def next_lower(self) -> "Scope":
```

Comment:
how often do we use this in lookup? i see us using those all over the time regenerating them all the time

off hand i cant tell of we have this in inner loops, but i'd really like to keep it out of loops

in particular list(Scopes) looks like something we should just make a constant
same goes for the index lookups

Reply:
Very rarely I think, but initially I had those implemented as external utility functions and cached. I'm happy to implement it like that again, if you agree.

Reply:
Done slightly different than the other time, let me know what you think.